.gitlab-ci.yml

```diff
@@ -18,3 +24,32 @@ test:unit:
     - git submodule update --init --recursive
   script:
     - make test
+
+test:integration:
+  stage: test
+  needs: []
+  image: docker:18-dind
+  tags:
+    - mender-qa-slave
+  before_script:
+    - /usr/local/bin/dockerd-entrypoint.sh &
+    - sleep 10
+    - export DOCKER_HOST="unix:///var/run/docker.sock"
+    - docker version
+    - export
+    # Install test dependencies
+    - apk --update --no-cache add bash gcc make curl util-linux openssh
+      python3 python3-dev libc-dev libffi-dev openssl-dev libc6-compat git
+    - CRYPTOGRAPHY_DONT_BUILD_RUST=1 pip3 install -r tests/integration/requirements.txt
+    - pip3 install docker-compose==1.24.0
+    # Init the git submodules
+    - git submodule update --init --recursive
```

Comment:
Note: if we go the way I suggested in my previous comment, this line needs to be done before the pip install/ apk add